### PR TITLE
Filter dust from stable supply changes

### PR DESCRIPTION
**Pull request by chapati23**
## The Problem

- The `/stables` supply changes table was cluttered with tiny mint/burn rows.
- Many of those rows rendered as `0.00`, making meaningful supply movements harder to scan.

## The Solution

- Hide supply-change rows whose absolute token-native amount is below `0.01 token`.
- Show the threshold directly in the table so operators know why tiny rows are omitted.

## Details

- Adds a shared threshold helper that scales `0.01` by each row's token decimals and applies the same absolute threshold to mints and burns.
- Fetches up to three 400-row raw pages when dust-heavy pages do not produce enough visible rows, then keeps the visible table capped at 200 rows.
- Preserves the existing capped warning when the fetched raw pages may hide older rows.
- Renders already-visible rows while follow-up pages keep loading instead of replacing them with the loading card.
- Adds focused helper, hook, and table tests for the threshold boundary, pagination, follow-up failures, background loading, and capped empty state.

## Validation

- `pnpm --filter @mento-protocol/ui-dashboard test -- stables` — passed, 2,885 tests.
- `pnpm agent:quality-gate --run` — passed all mapped dashboard commands.
- Pre-push `pnpm agent:quality-gate --run` — passed via fresh successful gate reuse.
- `pnpm agent:autoreview` — deterministic fallback was clean; Codex review engine was unavailable in this environment.
- Production-build fixture-backed browser smoke on `/stables` with Playwright/Chromium — passed: rendered `0.01 token` copy, displayed `+0.01`, `+0.02`, and `-0.03` rows, hid dust rows, requested GraphQL offsets `0`, `400`, and `800`, and had no console errors.
- chrome-devtools MCP was unavailable because its Chrome profile was locked, so browser verification used the repo's Playwright/Chromium tooling.

## Ship Checklist

- [x] ship skill used
- [x] local review/gate run
- [x] PR readiness probe planned

---

> [!NOTE]
> **Medium Risk**
> Changes GraphQL pagination and client-side filtering for the stables changes feed (up to three sequential queries), which can affect load and what operators see, but stays read-only dashboard logic with bounded fetch limits.
>
> **Overview**
> The `/stables` supply changes table now **drops dust rows** whose absolute token-native amount is below **0.01 token** (scaled by decimals, same rule for mints and burns), matching two-decimal display so operators no longer see pages of `0.00` noise.
>
> **`useStablesChanges`** filters raw GraphQL results through the new helpers, keeps the table capped at **200 visible rows**, and **conditionally loads up to three 400-row pages** when early pages are mostly filtered out. Loading, error, and **capped** behavior are adjusted so partial results still render and truncation is signaled when the fetch budget or follow-up pages may hide older events.
>
> The **Supply changes** table copy now states the threshold in empty and non-empty states (including capped empty). Tests cover the threshold math, hook pagination/filtering edge cases, and the capped empty UI.
>
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 671841518298c2fa0a5553fb73392a9faaed7ce1. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: ui-dashboard/src/app/stables/_lib/use-stables-data.ts [hunk 1]
```diff
@@ -137,21 +138,35 @@ export function useStablesCustodyDailySnapshots(_range: RangeKey) {
...
-    limit: CHANGES_PAGE_LIMIT,
-    offset: page * CHANGES_PAGE_LIMIT,
+    limit: CHANGES_QUERY_PAGE_LIMIT,
+    offset: page * CHANGES_QUERY_PAGE_LIMIT,
   });
-  const events = useMemo(() => data?.StableSupplyChangeEvent ?? [], [data]);
+  const rawEvents = useMemo(() => data?.StableSupplyChangeEvent ?? [], [data]);
+  const events = useMemo(() => {
+    const visibleEvents = rawEvents.filter(isVisibleSupplyChangeEvent);
+    return visibleEvents.slice(0, CHANGES_DISPLAY_LIMIT);
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/app/stables/_lib/use-stables-data.ts]: Fixed in d64809f7 — changed useStablesChanges to fetch additional raw pages when earlier pages are dust-heavy, so visible >=0.01 rows are not starved; browser smoke confirmed offsets 0, 400, and 800 were requested.

File: ui-dashboard/src/app/stables/_lib/use-stables-data.test.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      }),
+    );
+    expect(result.events.map((event) => event.id)).toEqual([
+      "visible-mint",
+      "visible-burn",
+    ]);
+    expect(result.capped).toBe(false);
+  });
+
+  it("keeps the capped warning when the fetched raw page may hide older rows", () => {
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/app/stables/_lib/use-stables-data.test.tsx]: Fixed in d64809f7 — added a 201-visible-event fixture that asserts the table returns 200 rows and keeps capped=true when visible rows exceed the display limit.

File: ui-dashboard/src/app/stables/_lib/use-stables-data.ts [hunk 3]
```diff
@@ -137,21 +138,35 @@ export function useStablesCustodyDailySnapshots(_range: RangeKey) {
...
+  }, [rawEvents]);
+  const capped = useMemo(() => {
+    if (rawEvents.length === CHANGES_QUERY_PAGE_LIMIT) return true;
+    let visibleCount = 0;
+    for (const event of rawEvents) {
+      if (isVisibleSupplyChangeEvent(event)) visibleCount += 1;
+      if (visibleCount > CHANGES_DISPLAY_LIMIT) return true;
+    }
+    return false;
+  }, [rawEvents]);
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/app/stables/_lib/use-stables-data.ts]: Fixed in d64809f7 — split the hook into per-page helpers so each fetched raw page is filtered once, then combined by buildVisibleChangesResult.

File: ui-dashboard/src/app/stables/_lib/aggregate.ts [hunk 4]
```diff
@@ -7,12 +7,31 @@ import type { OracleRateMap } from "@/lib/tokens";
...
   RangeKey,
   StableSupplyDailySnapshot,
+  StableSupplyChangeEvent,
   StableTokenCustodyDailySnapshot,
   TokenAgg,
 } from "./types";
 
 const SECONDS_PER_DAY = BigInt(86_400);
 const SECONDS_PER_DAY_NUMBER = 86_400;
+export const SUPPLY_CHANGE_FILTER_THRESHOLD_LABEL = "0.01 token";
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/app/stables/_lib/aggregate.ts:19]: Fixed in d64809f7 — added a co-located comment tying SUPPLY_CHANGE_FILTER_THRESHOLD_LABEL to minimumVisibleSupplyChangeRaw and the 0.01 = 1/100 token threshold.

**Comment by chapati23** [comment]
Fixed in d64809f7 — addressed the Claude review batch: multi-page dust-heavy fetching, display-limit capped coverage, and threshold-label drift comment. I also replied to each inline thread.

File: ui-dashboard/src/app/stables/_lib/use-stables-data.ts [hunk 5]
```diff
@@ -137,21 +146,120 @@ export function useStablesCustodyDailySnapshots(_range: RangeKey) {
...
+  }
+  return null;
+}
+
+function firstEnabledPageError(pages: ReadonlyArray<ChangePageState>): unknown {
+  for (const candidate of pages) {
+    if (candidate.enabled && candidate.error != null) return candidate.error;
+  }
+  return null;
+}
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/app/stables/_lib/use-stables-data.ts:273]: Fixed in 8cd0c93a — follow-up page errors now degrade open when earlier pages already produced visible rows: the hook keeps those rows, clears the fatal table error, and marks the result capped; added a regression test for that case.

File: ui-dashboard/src/app/stables/_lib/use-stables-data.ts [hunk 6]
```diff
@@ -137,21 +146,136 @@ export function useStablesCustodyDailySnapshots(_range: RangeKey) {
...
+  const { events, capped } = buildVisibleChangesResult(pages, pageError);
   return {
     events,
-    error,
-    isLoading,
-    capped: events.length === CHANGES_PAGE_LIMIT,
+    error: visibleChangesError(firstPage, events, pageError),
+    isLoading: pages.some(
+      (candidate) => candidate.enabled && candidate.isLoading,
+    ),
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/app/stables/_lib/use-stables-data.ts]: Fixed in 67184151 — visible first-page supply changes now render while follow-up pages continue loading, and the result stays capped during background pagination.
